bochs: refactor

AndersonTorres opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

AndersonTorres committed:
bochs: move pins to top-level

wegank commented on pkgs/by-name/bo/bochs/package.nix:
```diff
@@ -143,8 +143,10 @@ stdenv.mkDerivation (finalAttrs: {
+    mainProgram = "bochs";
     maintainers = with lib.maintainers; [ AndersonTorres ];
     platforms = lib.platforms.unix;
+    broken = stdenv.isDarwin;
```
The package isn't broken on aarch64-darwin, however? https://hydra.nixos.org/job/nixpkgs/trunk/bochs.aarch64-darwin

AndersonTorres replied:
Still broken on Darwin ARM

Qyriad replied:
it's certainly still broken on aarch64-darwin for me
[z1xcamxmlhndf8vrzsk2gknrm4whqdvl-bochs-2.8.drv.log](https://github.com/NixOS/nixpkgs/files/15288092/z1xcamxmlhndf8vrzsk2gknrm4whqdvl-bochs-2.8.drv.log)

AndersonTorres replied:
OK, then I will break it on Darwin, and wait for the Darwin guys.

AndersonTorres committed:
bochs: refactor

- fetch from GitHub instead of SourceForge
- use lib.* functions
- mainProgram
- split outputs
- strictDeps

AndersonTorres committed:
bochs: yet another workaround for Darwin

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/call-for-discussion-why-and-how-much-should-we-care-about-the-override-interfaces-part-i-duckstation-and-pcsx2/45117/1

AndersonTorres commented:
@wegank

wegank commented on pkgs/by-name/bo/bochs/package.nix:
```diff
@@ -60,80 +67,86 @@ stdenv.mkDerivation (finalAttrs: {
+    (lib.enableFeature (!stdenv.isDarwin) "sb16")
   ];
 
+  outputs = [ "out" "lib" "doc" "man" ];
```
```suggestion
  outputs = [ "out" "doc" "man" ] ++ lib.optionals (!stdenv.isDarwin) [ "lib" ];
```

For unbreaking on aarch64-darwin.
